scripts/lib/CIME/SystemTests/pet.py

```diff
@@ -30,6 +30,10 @@ def _case_one_setup(self):
             if self._case.get_value("NTHRDS_%s"%comp) <= 1:
                 self._case.set_value("NTHRDS_%s"%comp, 2)
 
+        # Subtle: machines like titan will not let you overload the number of threads
```

Comment:
In pet.py _case_one_setup you set NTHRDS = 2 for each component **then you run case_setup to recreate the case.run and case.test scripts for the correct node and task setup**
In _case_two_setup you set NTHRDS = 1 and MAX_TASKS_PER_NODE to half it's original value 
and run case.setup again to recreate the case.run and case.test scripts for the same number of nodes but undersubscribed.    I think that the issue that you are having is that the aprun command line is not getting updated to be consistent with these changes.